### PR TITLE
[#81] Alternative approach to fix #81

### LINKED ISSUE #81: Incorrect format when macro is used while building a binary
I was able to successfully reformat quite a big codebase and have to say only one thing - What a great job!

The only issue that I found is that I had multiple syntax issues with this pattern:
```erlang
<<?i2b(IntegerValue)/binary, " foobar">>
```
instead of correct one:
```erlang
<<(?i2b(IntegerValue))/binary, " foobar">>
```
`?i2b/1` macro is just a conversion from integer to binary using BIF function without any wrapper. 

Thank you again for a great tool! :pray:

### PR DESCRIPTION
**Pull request by elbrujohalcon**
Another way to fix #81, an alternative to #257.

### CODE REVIEW

File: test_app/src/macros_in_binary.erl [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+-export([chunkify/1, p/0]).
+
+-define(CHUNK_SIZE, 65535 - 1).
+-define(CONCAT(A, B), <<A/binary, B/binary>>).
+-define(A_MACRO(X), X + 1).
+-define(A_FUNCTION, rand:uniform).
+-define(NO_PARENS, <<"no_parens">>).
+
+chunkify(<<Binary:?CHUNK_SIZE/binary, Rest/binary>>) ->
+    {<<(?CONCAT(Binary, Rest))/binary>>, <<(?CONCAT(Binary, Rest))/binary>>}.
```
Comment (vkatsuba) [inline test_app/src/macros_in_binary.erl:12]: Could you please change:
```erlang
{<<(?CONCAT(Binary, Rest))/binary>>, <<(?CONCAT(Binary, Rest))/binary>>}.
```
to
```erlang
{<<?CONCAT(Binary, Rest)/binary>>, <<?CONCAT(Binary, Rest)/binary>>}.
```
?

Comment (elbrujohalcon) [reply test_app/src/macros_in_binary.erl:12]: Oh, right… my bad. Yeah.

Comment (elbrujohalcon) [reply test_app/src/macros_in_binary.erl:12]: Good catch!

**Comment by elbrujohalcon** [comment]
@vkatsuba what you think about this approach now?

**Comment by vkatsuba** [comment]
> @vkatsuba what you think about this approach now?

Looks pretty impressive :muscle:

**Review by vkatsuba — APPROVED** [review]

**Comment by elbrujohalcon** [comment]
That won't be the output, @vkatsuba 
The output will be `<<(?LIST_TO_BINARY("Was_List"))/binary>>`, right?
Because the formatter can't tell if the macro will end up writing code that needs to go between parentheses or not… so it will preemptively parenthesize the thing.
This is exactly what the example does with `?CONCAT/2`.
I don't think that your particular example will add much, unless I'm missing something.

BTW… your intended macro definition is this one, right? …

```erlang
-define(LIST_TO_BINARY(X), list_to_binary(X)).
```

**Comment by elbrujohalcon** [comment]
> I see, but this expected behavior I suppose will need to describe somewhere, eg examples of input/output or just provide links into exist in/out modules for comparing

Oh, right! Yeah… that would be lovely… but we don't have that for any formatting rule whatsoever. And we inherited lots of formatting decisions from the OTP team, anyway.
We might eventually document them all, but I don't see the point in thoroughly documenting just one.

**Comment by vkatsuba** [comment]
> We might eventually document them all, but I don't see the point in thoroughly documenting just one.

Yep, I was mean - all possible examples what already provided in repo, just for easy navigation. But this is just a suggestion.

**Pull request merged**